models/facility.py

```diff
@@ -130,11 +140,105 @@ def remove_staff(self, old_staff):
         Return True if succesfull else False
         """
         if old_staff.id in self.person['staff']:
+            self.delete_from_db(old_staff)
             del self.person['staff'][old_staff.id]
             return True
         return False
 
+    def save_state(self, database_name="default.db", over_write=False, up=" "):
+        if type(database_name) != str:
+            raise TypeError
+        if over_write:
+            if os.path.exists("models/database/" + database_name):
+                os.remove("models/database/" + database_name)
+            raise OverWriteException
+        if os.path.exists("models/database/" + database_name) and up:
+            raise UpdateException
```

Comment:
I can see in this script you raise various exceptions which is ok but they should not terminate the program they should instead give a relevant message to the user so that they can perform any remedial actions that you could explicitly prescribe